### DISCUSSION TITLE
Changed settings revert back after a few seconds

### DISCUSSION THREAD

**Discussion by willbiker** [wills106/homeassistant-solax-modbus#680] [General]
Hi all. I have a Solax G4 hybrid inverter and when I change a setting in Home assistant, it looks as if the setting change is applied but then it reverts back 5 seconds later. Ive enabled debugging to see whats going on and I can see a notification stating that the setting has 'become unknown'. The weird thing is this revert only seems to happen some of the time, other times it writes setting changes ok. Any ideas? Thank you

**Comment by willbiker** [comment]
![Screenshot_20231227_183041_Home Assistant](https://github.com/wills106/homeassistant-solax-modbus/assets/117373918/f97061f7-eedc-4e59-9740-cfd1cbbed654)

**Comment by cleveHEX** [comment]
Read docs and faq.

**Comment by willbiker** [comment]
For others who may also hit this issue, it Looks like it could be my inverter going into idle when solar generation is 0 and battery is empty...I will try some automation to switch to manual mode at 12% SOC in order to keep it awake.

**Reply by redsand451** [reply]
I have the same question for Solax G4 X1 hybrid if anyone can enlighten me please (only been using HA for a week!)

I have set up automations to control charge period 2 when Octopus Intelligent smart schedules are outside the normal off peak period (charge period 1 already set start 23:30 to end 05:30 on Solax).  Automation worked last night at 23:00 to start charge period 2 when an extra Smart schedule was available and then switched it off at 23:30 when the normal off peak period started. The charge period 2 enable/disable updated and remained but the start/end time fields updated and then quickly reverted back to original value.  Same thing happened when I later tried to update manually through HA.

It is not the end of the world, as I can keep the start/end times fixed from 00:00 to 23:59 for charge period 2 and just disable/enable as required.  I was trying to also set charge period 2 end to 00:00 to double down on disabling it. 

Battery capacity is well above 10%.  SolaX Device Lock = unlocked and Solax Lock State =  Unlocked - Advanced.

Out of curiosity, what is reverting the values back and why? Is there something else I am missing to update these fields permanently (not necessarily for this automation but potentially for future ones).

And noted on the point in FAQs that unable to control Inverter when PV = 0 & Battery SOC is at Minimum. So is the answer to set an automation to monitor the Battery SOC and at e.g. 11% enable charge for a short period of time? Any recommendations?

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/154935950/331e6d55-cec3-4574-b128-b7d52a6a9e56)

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/154935950/eb4c5484-dd21-4952-b719-bdd1a775cab2)

**Reply by redsand451** [reply]
Just had a thought on the first bit ... the Charger 2 start/end time are only visible in the app when the charge period 2 is enabled. So does the same apply under HA automation updates,  i.e. has to be enabled first to update the times and/or then disable as required.
Tried this via HA manually (enable, update time, disable) and now the time has kept the value with no revert.

@willbiker is your issue with Selfuse Night Charge Enable needing to be set before you update Charger Start Time 1?

**Comment by wills106** [comment]
> it looks as if the setting change is applied but then it reverts back 5 seconds later.

I am assuming your poll rate is 5s

So the Integration attempts to write. The Inverter ignores the write command due to the fact the Inverter is "sleeping" if then reverts back to whatever the previous value was on the next 5s poll.

TOU registers may help? Need to implement them though and you will need to be on a latest Firmware.
